x-pack/legacy/plugins/uptime/public/state/actions/ui.ts

```diff
@@ -3,53 +3,19 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-
-export const SET_INTEGRATION_POPOVER_STATE = 'SET_INTEGRATION_POPOVER_STATE';
-export const SET_BASE_PATH = 'SET_BASE_PATH';
-export const REFRESH_APP = 'REFRESH_APP';
+import { createAction } from 'redux-actions';
 
 export interface PopoverState {
   id: string;
   open: boolean;
 }
 
-interface SetBasePathAction {
-  type: typeof SET_BASE_PATH;
-  payload: string;
-}
-
-interface SetIntegrationPopoverAction {
-  type: typeof SET_INTEGRATION_POPOVER_STATE;
-  payload: PopoverState;
-}
-
-interface TriggerAppRefreshAction {
-  type: typeof REFRESH_APP;
-  payload: number;
-}
+export type UiPayload = PopoverState & string & number & Map<string, string[]>;
 
-export type UiActionTypes =
-  | SetIntegrationPopoverAction
-  | SetBasePathAction
-  | TriggerAppRefreshAction;
+export const setBasePath = createAction<string>('SET BASE PATH');
+export const triggerAppRefresh = createAction<number>('REFRESH APP');
 
-export function toggleIntegrationsPopover(popoverState: PopoverState): SetIntegrationPopoverAction {
-  return {
-    type: SET_INTEGRATION_POPOVER_STATE,
-    payload: popoverState,
-  };
-}
-
-export function setBasePath(basePath: string): SetBasePathAction {
-  return {
-    type: SET_BASE_PATH,
-    payload: basePath,
-  };
-}
-
-export function triggerAppRefresh(refreshTime: number): TriggerAppRefreshAction {
-  return {
-    type: REFRESH_APP,
-    payload: refreshTime,
-  };
-}
+export const setFilters = createAction<Map<string, string[]>>('SET FILTERS');
+export const toggleIntegrationsPopover = createAction<PopoverState>(
+  'TOGGLE INTEGRATION POPOVER STATE'
+);
```

Comment:
@justinkambic do you have any comments about reducing boiler plate code in this file?

Reply:
I think it looks great, much cleaner. I think it's about as close to entropy as we can expect at this point. Did you have any specific concern?